cloud_chatops/src/pr_reminder.py

```diff
@@ -0,0 +1,205 @@
+"""This module handles the posting of messages to Slack using the Slack SDK WebClient class."""
+
+from typing import List, Dict
+from datetime import datetime, timedelta
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+from src.read_data import get_token, get_user_map, get_repos
+from src.get_github_prs import GetGitHubPRs
+
+
+class PostPRsToSlack:
+    """
+    This class handles the Slack posting.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.repos = get_repos()
+        self.client = WebClient(token=get_token("SLACK_BOT_TOKEN"))
+        self.slack_ids = get_user_map()
+        self.prs = GetGitHubPRs(get_repos(), "stfc").run()
+        self.channel = "temp-chatops"
+
+    def run(self, mention, channel=None) -> None:
+        """
+        This method calls the functions to post the reminder message and subsequent PR messages.
+        :param channel: Changes the channel to post the messages to.
+        :param mention: To mention the users in Slack or just post their name.
+        """
+        if channel:
+            self.channel = channel
+        reminder_message = self.post_reminder_message()
+        self.post_thread_messages(self.prs, reminder_message, mention)
+
+    def post_reminder_message(self) -> WebClient.chat_postMessage:
+        """
+        This method posts the main reminder message to start the thread if PR notifications.
+        :return: The reminder message return object
+        """
+        reminder = self.client.chat_postMessage(
+            channel=self.channel,
+            text="Here are the outstanding PRs as of today:",
+        )
+        return reminder
+
+    def post_thread_messages(
+        self,
+        prs: Dict[str, List],
+        reminder_message: WebClient.chat_postMessage,
+        mention: bool,
+    ) -> None:
+        """
+        This method iterates through each PR and calls the post method for them.
+        :param mention: To mention the users or not
+        :param prs: A list of PRs from GitHub
+        :param reminder_message: The reminder message object
+        """
+        prs_found = False
+        for repo in prs.values():
+            for pr in repo:
+                info = {
```

Comment:
if we set `GetGitHubPRs.run()` to output a list of dataclasses with the information already parsed into the way we want we could potentially remove the need for this